submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+# Support fine-grained cluster connectivity and multiple cable drivers
+
+## Summary
+
+Submariner currently assumes full mesh connectivity by default and provisions VPN connections between all participating
+clusters. In addition, Submariner currently offers one of three cable drivers: VXLAN, IPSec or wireguard with limited
+configuration options. Only one of these cable drivers is enabled across all the clusters in a cluster set, that is all
+clusters that join a broker must use the same cable driver. The goal of this proposal is to outline an extended model
+that will support:
+
+- The selection of which clusters to connect and in what fashion (full mesh or connection to subset of clusters in
+  a ClusterSet).
+- The selection of one or more cable drivers in order to support different cable types between connected clusters.
+  For example, IPSec between public and private clouds but raw VXLAN between the clusters in the same cloud.
+
+There are use cases that involve supporting a mix of cable drivers or cable driver configurations within the same
+cluster set. For example, a cluster set could contain three clusters, two that are on-premise and the other
+in the cloud. Connections to the cloud cluster will need to use encryption while the on-premise clusters can use
+unencrypted connections, as they’re connected by a private link, to avoid the performance overhead of encryption.
+This could be achieved using either:
+
+- Different cable driver implementations or
+- The same cable driver implementation with differing configurations.
+
+For other use cases, it makes sense to offer finer-grained connectivity options. One such use case is a client-server
+scenario where multiple client clusters need to access a server cluster but the user does not want the client clusters
+to connect to one another. Finer-grained connectivity could also improve overall scalability where full mesh is not
+required.
+
+## Proposal
+
+This enhancement proposes a mechanism to allow users to specify their intent as to which clusters to connect and how to
```

Comment:
This proposal allows defining a list of 'neighbors' clusters for each cluster and how to connect (vxlan,ipsec, etc)  to each 'neighbor'.

It isn't clear to me how the entire clusters connectivity is derived from this information. 

for example, in the diagram you attached, do you expect cluster#2 will be able to establish VPN with cluster#6 though they aren't 'neighbors'?

Reply:
Got it, thanks for the clarification @maryamtahhan .

I can see the added value in allowing flexibility in SM topology I think the current proposal allows high flexibility in topology configuration, which I am not sure all of that is required in the context of Submariner.

Maybe it would be useful to add a section with the requirements we would like to cover in the context of SM topology config flexibility, what do you think?

This section may help to clarify things like:
- Do we want to support different cable types between connected clusters (like in your example with IpSec and VxLAN)? 
- Do we want to support multiple cable drivers of the same type (e.g: unencrypted) on topology?  
- Do we want to support directional granularity?  IIUC, the current proposal allows to define for example A-->B will connect via VxLan and reverse direction (B --> A) will use IpSec
- How does partially connected topology comply with MCS service export API?
- etc